packages/customize-widgets/src/components/header/index.js

```diff
@@ -1,25 +1,69 @@
 /**
  * WordPress dependencies
  */
-import { createPortal } from '@wordpress/element';
-import { __, _x } from '@wordpress/i18n';
+import { createPortal, useState, useEffect } from '@wordpress/element';
+import { __, _x, isRTL } from '@wordpress/i18n';
 import { Button, ToolbarItem } from '@wordpress/components';
 import { NavigableToolbar } from '@wordpress/block-editor';
-import { plus } from '@wordpress/icons';
+import { displayShortcut } from '@wordpress/keycodes';
+import { plus, undo as undoIcon, redo as redoIcon } from '@wordpress/icons';
 
 /**
  * Internal dependencies
  */
 import Inserter from '../inserter';
 
-function Header( { inserter, isInserterOpened, setIsInserterOpened } ) {
+function Header( {
+	sidebar,
+	inserter,
+	isInserterOpened,
+	setIsInserterOpened,
+} ) {
+	const [ [ hasUndo, hasRedo ], setUndoRedo ] = useState( [
+		sidebar.hasUndo(),
+		sidebar.hasRedo(),
+	] );
+
+	useEffect( () => {
+		return sidebar.subscribeHistory( () => {
```

Comment:
Not sure I understand? We are subscribing in the effect itself, but also returning the **unsubscribe** callback to the cleanup callback. It's a common pattern for event listeners.

Reply:
What confused me is it's not clear at a glance what `sidebar.subscribeHistory` does. the usual pattern for effects is something like:
```
useEffect( () => {
  subscribe()
  return () => unsubscribe()
})
``` 
but in this case, both the `subscribe()` and the returning of the cleanup function are happening in `sidebar.subscribeHistory`.

Reply:
I'd argue that this pattern is in fact very common. As both Redux's [`store.subscribe`](https://redux.js.org/api/store#subscribelistener) and history's [`listen`](https://github.com/ReactTraining/history/blob/master/docs/api-reference.md#historylistenlistener-listener) are using this pattern to return an unsubscribe function.

However, I agree that it never hurts to make it more clear. I'll try to keep that in mind next time :)

Reply:
Yeah, after digging in a bit I realised it occurred in other places. Was just surprising the first time 😅 
I guess depending on how complex the logic is in the effect it might help to add a comment, but in this case it's pretty straightforward.